common/src/paths.rs

```diff
@@ -0,0 +1,23 @@
+use std::path::PathBuf;
+
+pub const QUALIFIER: &'static str = "org";
+pub const ORGANIZATION: &'static str = "holochain";
+pub const APPLICATION: &'static str = "holochain";
+pub const KEYS_DIRECTORY: &'static str = "keys";
+
+/// Returns the path to the root config directory for all of Holochain.
+/// If we can get a user directory it will be an XDG compliant path
+/// like "/home/peter/.config/holochain".
+/// If it can't get a user directory it will default to "/etc/holochain".
+pub fn config_root() -> PathBuf {
+    directories::ProjectDirs::from(QUALIFIER, ORGANIZATION, APPLICATION)
+        .and_then(|dirs| Some(dirs.config_dir().to_owned()))
+        .or(Some(PathBuf::new().join("/etc").join(APPLICATION)))
+        .unwrap()
```

Comment:
Better make this an `unwrap_or_else`:
```rust
unwrap_or_else(|| PathBuf::from("something"));
```
also the `/etc` path won't work on windows either

Reply:
I think we should bubble up the error instead of choosing a dummy path